optuna/trial.py

```diff
@@ -136,6 +136,8 @@ def __init__(
 
         self._init_relative_params()
 
+        self.distributions_in_trial = {}  # type: Dict[str, Any]
```

Comment:
You could be more precise.
```suggestion
        self.distributions_in_trial = {}  # type: Dict[str, Dict[str, Any]]
```